### PR TITLE
E UI 4716 hearing venue

### PR DESCRIPTION
**Pull request by waynepHMCTS**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/EUI-4716

### Change description ###
re-instate unit test, validation and add location methods and hearing venue route and component name change to correct page name.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/hearings/containers/request-hearing/hearing-venue/hearing-venue.component.ts [hunk 1]
```diff
@@ -41,29 +46,47 @@ export class HearingVenueComponent extends RequestHearingPageFlow implements OnI
...
+    this.findLocationFormGroup.controls.locationSelectedFormControl.setValue(undefined);
+    this.findLocationFormGroup.controls.locationSelectedFormControl.markAsPristine();
+    this.locationsFound$ = of([]);
+  }
+
   public removeSelection(location: LocationByEPIMSModel): void {
     this.selectedLocations$.subscribe(selectedLocations => {
       const index = selectedLocations.findIndex(selectedLocation => selectedLocation.epims_id === location.epims_id);
       selectedLocations.splice(index, 1);
     });
```
[the remark this answers is not in the document]
Comment (waynepHMCTS) [reply src/hearings/containers/request-hearing/hearing-venue/hearing-venue.component.ts:79]: I do not understand.

File: src/hearings/containers/request-hearing/hearing-venue/hearing-venue.component.ts [hunk 2]
```diff
@@ -41,29 +46,47 @@ export class HearingVenueComponent extends RequestHearingPageFlow implements OnI
...
   }
 
   public addSelection(): void {
     if (this.findLocationFormGroup.controls.locationSelectedFormControl.value) {
       this.selectedLocations$.subscribe(selectedLocations => {
-        selectedLocations.push(this.findLocationFormGroup.controls.locationSelectedFormControl.value as LocationByEPIMSModel);
-        this.findLocationFormGroup.controls.locationSelectedFormControl.setValue(undefined);
-        const strLocations = selectedLocations.map(location => location.region).join(',');
-        this.hearingStore.dispatch(new fromHearingStore.SaveHearingConditions({region: strLocations}));
+          this.appendLocation(selectedLocations);
```
[the remark this answers is not in the document]
Comment (waynepHMCTS) [reply src/hearings/containers/request-hearing/hearing-venue/hearing-venue.component.ts:59]: I do not understand please give an example off code.

[the remark this answers is not in the document]
Comment (rayliangatos) [reply src/hearings/containers/request-hearing/hearing-venue/hearing-venue.component.ts:59]: explained on https://tools.hmcts.net/jira/browse/EUI-4952

File: src/hearings/containers/request-hearing/hearing-venue/hearing-venue.component.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (rayliangatos) [reply src/hearings/containers/request-hearing/hearing-venue/hearing-venue.component.ts:78]: explained on https://tools.hmcts.net/jira/browse/EUI-4952

File: src/hearings/containers/request-hearing/hearing-venue/hearing-venue.component.spec.ts [hunk 3]
```diff
@@ -1,32 +1,50 @@
...
+  public autoCompleteInputBox: MockAutoCompleteInputBox = new MockAutoCompleteInputBox();
 }
 
 describe('HearingVenueComponent', () => {
   let component: HearingVenueComponent;
   let fixture: ComponentFixture<HearingVenueComponent>;
   const mockedHttpClient = jasmine.createSpyObj('HttpClient', ['get', 'post']);
   const hearingsService = new HearingsService(mockedHttpClient);
-  hearingsService.navigateAction$ = of(ACTION.CONTINUE);
+ // hearingsService.navigateAction$ = of(ACTION.CONTINUE);
```
Comment (rayliangatos) [inline src/hearings/containers/request-hearing/hearing-venue/hearing-venue.component.spec.ts:47]: should this removed

**Review by rayliangatos — COMMENTED** [review]
please fix this in advance
https://tools.hmcts.net/jira/browse/EUI-4951

**Review by rayliangatos — CHANGES_REQUESTED** [review]
Request changes:
1. cause issue in https://tools.hmcts.net/jira/browse/EUI-4951
2. page flow is broken

File: src/hearings/containers/request-hearing/hearing-venue/hearing-venue.component.ts [hunk 4]
```diff
@@ -16,23 +16,27 @@ import {RequestHearingPageFlow} from '../request-hearing.page.flow';
...
-  constructor(protected readonly hearingStore: Store<fromHearingStore.State>,
-              protected readonly hearingsService: HearingsService,
-              fb: FormBuilder) {
+  constructor(public readonly hearingStore: Store<fromHearingStore.State>, fb: FormBuilder,
+              protected readonly hearingsService: HearingsService) {
     super(hearingStore, hearingsService);
-    this.findLocationFormGroup = fb.group({
-      locationSelectedFormControl: [null]
+    this.findLocationFormGroup =  fb.group({
+      locationSelectedFormControl: [null, Validators.required]
```
Comment (rayliangatos) [inline src/hearings/containers/request-hearing/hearing-venue/hearing-venue.component.ts:35]: from business perspective under this hearing value page, this control shouldn't be required.

**Pull request merged**